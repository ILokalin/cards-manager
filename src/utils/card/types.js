export const STUDENT_CARD_TYPE = "STUDENT";
export const COVER_CARD_TYPE = "COVER";

export const emptyCardTemplate = {
  id: "_00000000",
  type: COVER_CARD_TYPE,
  firstName: "Имя",
  lastName: "Неустановлено",
};

const CardTypes = {
  [STUDENT_CARD_TYPE]: { genitive: "студента" },
  DEFAULT: { genitive: "неустановленного типа" },
};

export const getCardType = (card) => {
  return {
    genitive: CardTypes[card?.type]?.genitive ?? CardTypes.DEFAULT.genitive,
  };
};
