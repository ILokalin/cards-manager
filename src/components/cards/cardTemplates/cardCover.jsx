import { CardCommon } from "./cardCommon";

export const CardCover = (props) => {
  return (
    <CardCommon {...props}>
      <p className="card-text mt-4">Нет данных для генерации карты</p>
    </CardCommon>
  );
};
