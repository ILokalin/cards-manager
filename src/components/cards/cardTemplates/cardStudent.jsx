import { CardCommon } from "./cardCommon";

export const CardStudent = (props) => {
  const { course, college } = props;
  return (
    <CardCommon {...props}>
      <p className="card-text mt-4">
        Учебное заведение:&nbsp;
        <span className="fw-bold">{college}</span>
      </p>
      <p className="card-text">
        Студент:&nbsp;
        <span className="fw-bold">{course}-го курса</span>
      </p>
    </CardCommon>
  );
};
