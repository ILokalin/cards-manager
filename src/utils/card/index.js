export {
  STUDENT_CARD_TYPE,
  COVER_CARD_TYPE,
  getCardType,
  emptyCardTemplate,
} from "./types.js";
export { CardFactory } from "./cardFactory.js";
