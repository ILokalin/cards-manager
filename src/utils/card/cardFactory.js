import { STUDENT_CARD_TYPE, COVER_CARD_TYPE } from "./types";
import { CardStudent, CardCover } from "components/cards/cardTemplates";

export const CardFactory = {
  [STUDENT_CARD_TYPE]: (cardProps) => <CardStudent {...cardProps} />,
  [COVER_CARD_TYPE]: (cardProps) => <CardCover {...cardProps} />,
};
