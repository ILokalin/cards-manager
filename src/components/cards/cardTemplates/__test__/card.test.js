import { render } from "@testing-library/react";
import { Card } from "../card";
import { CardCover } from "../cardCover";
import { CardStudent } from "../cardStudent";

describe("Cards", () => {
  it("Basic card is rendered", () => {
    const { asFragment } = render(<Card />);
    const snapshot = asFragment();
    expect(snapshot).toMatchSnapshot();
  });

  it("CardsCover is rendered", () => {
    const { asFragment } = render(<CardCover />);
    const snapshot = asFragment();
    expect(snapshot).toMatchSnapshot();
  });

  it("CardsStudent is rendered", () => {
    const { asFragment } = render(<CardStudent />);
    const snapshot = asFragment();
    expect(snapshot).toMatchSnapshot();
  });
});
