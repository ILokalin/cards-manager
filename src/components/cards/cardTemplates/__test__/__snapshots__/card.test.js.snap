// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cards CardsCover is rendered 1`] = `
<DocumentFragment>
  <article
    class="card p-4"
  >
    <h3
      class="card-title"
    >
       
    </h3>
    <p
      class="card-text mt-4"
    >
      Нет данных для генерации карты
    </p>
  </article>
</DocumentFragment>
`;

exports[`Cards CardsStudent is rendered 1`] = `
<DocumentFragment>
  <article
    class="card p-4"
  >
    <h3
      class="card-title"
    >
       
    </h3>
    <p
      class="card-text mt-4"
    >
      Учебное заведение: 
      <span
        class="fw-bold"
      />
    </p>
    <p
      class="card-text"
    >
      Студент: 
      <span
        class="fw-bold"
      >
        -го курса
      </span>
    </p>
  </article>
</DocumentFragment>
`;

exports[`Cards CardsTemplate is rendered 1`] = `
<DocumentFragment>
  <article
    class="card p-4"
  >
    <h3
      class="card-title"
    >
       
    </h3>
  </article>
</DocumentFragment>
`;
